src/dusk/serde_support.rs

```diff
@@ -0,0 +1,124 @@
+extern crate alloc;
+
+use alloc::string::String;
+
+use serde::{de::Error, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{AffinePoint, ExtendedPoint};
+
+impl Serialize for AffinePoint {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        let s = hex::encode(self.to_bytes());
+        serializer.serialize_str(&s)
+    }
+}
+
+impl<'de> Deserialize<'de> for AffinePoint {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        let s = String::deserialize(deserializer)?;
+        let decoded = hex::decode(&s).map_err(Error::custom)?;
+        let decoded_len = decoded.len();
+        let bytes: [u8; 32] = decoded
+            .try_into()
+            .map_err(|_| Error::invalid_length(decoded_len, &"32"))?;
+        AffinePoint::from_bytes(bytes)
+            .into_option()
+            .ok_or(Error::custom(
+                "Failed to deserialize AffinePoint: invalid AffinePoint",
+            ))
+    }
+}
+
+impl Serialize for ExtendedPoint {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        AffinePoint::from(self).serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for ExtendedPoint {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        AffinePoint::deserialize(deserializer).map(Into::into)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use group::Group;
+
+    use crate::{AffinePoint, ExtendedPoint, Fr};
+
+    #[test]
+    fn affine_point() {
+        let point = AffinePoint::identity();
+        let ser = serde_json::to_string(&point).unwrap();
+        let deser = serde_json::from_str(&ser).unwrap();
+        assert_eq!(point, deser);
+    }
+
+    #[test]
+    fn extended_point() {
+        let mut rng = rand_core::OsRng;
+        let point = ExtendedPoint::random(&mut rng);
+        let ser = serde_json::to_string(&point).unwrap();
+        let deser = serde_json::from_str(&ser).unwrap();
+        assert_eq!(point, deser);
+    }
+
+    #[test]
+    fn wrong_encoded() {
+        let wrong_encoded = "wrong-encoded";
+
+        let affine_point: Result<AffinePoint, _> =
+            serde_json::from_str(&wrong_encoded);
+        assert!(affine_point.is_err());
+
+        let extended_point: Result<ExtendedPoint, _> =
+            serde_json::from_str(&wrong_encoded);
+        assert!(extended_point.is_err());
+
+        let fr: Result<Fr, _> = serde_json::from_str(&wrong_encoded);
+        assert!(fr.is_err());
```

Comment:
The tests for `Fr` also needs to be in `fr::dusk`.
I wouldn't mind if you add them to the `serde_fr` function though if you add a comment on why they are expected to fail.